pyvista/core/dataset.py

```diff
@@ -2523,3 +2522,49 @@ def cell_type(self, ind: int) -> int:
 
         """
         return self.GetCellType(ind)
+
+    def point_is_inside_cell(self, ind: int, point: Vector) -> bool:
+        """Return whether a point is inside a cell.
+
+        Parameters
+        ----------
+        ind : int
+            Cell ID.
+
+        point : Sequence[float] or numpy.ndarray
+           Length 3 sequence defining a point in space.
+
+        Returns
+        -------
+        bool
+            Whether point is inside cell.
+
+        Examples
+        --------
+        >>> from pyvista import examples
+        >>> mesh = examples.load_hexbeam()
+        >>> mesh.cell_bounds(0)
+        (0.0, 0.5, 0.0, 0.5, 0.0, 0.5)
+        >>> mesh.point_is_inside_cell(0, [0.2, 0.2, 0.2])
+        True
+
+        """
+        if ind < 0 or ind >= self.n_cells:
+            raise ValueError(f"ind must be >= 0 and < {self.n_cells}, got {ind}")
+
+        if not isinstance(point, (collections.abc.Sequence, np.ndarray)):
+            raise ValueError(f"point must be a Sequence or ndarray, got {point}")
+
+        if len(point) != 3:
+            raise ValueError(f"point must have length 3, got {len(point)}")
+
+        cell = self.GetCell(ind)
+        npoints = cell.GetPoints().GetNumberOfPoints()
+
+        closest_point = [0.0, 0.0, 0.0]
+        sub_id = _vtk.mutable(0)
+        pcoords = [0.0, 0.0, 0.0]
+        dist2 = _vtk.mutable(0.0)
```

Comment:
If we don't use these mutables after the call, do they still have to be mutable? Do we get a segfault otherwise?

Reply:
A bit better than a segfault, but a `TypeError`:

```
E   TypeError: EvaluatePosition argument 3: a vtk.reference() object is required
```